compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6390,6 +6392,39 @@ private void reorderNamedArgs(BLangInvocation iExpr, BInvokableSymbol invokableS
         iExpr.requiredArgs = args;
     }
 
+    private void createIncludedRecordLiteral(BLangInvocation iExpr, BVarSymbol param, Map<String,
+            BLangExpression> namedArgs, List<BLangExpression> args) {
+        BLangRecordLiteral recordLiteral = TreeBuilder.createBLangRecordLiteralNode();
+        recordLiteral.type = param.type;
+        if (param == ((BInvokableSymbol) iExpr.symbol).incRecordParamAllowAdditionalFields) {
+            for (String name : namedArgs.keySet()) {
+                BLangNamedArgsExpression expr = (BLangNamedArgsExpression) namedArgs.get(name);
+                if (!((BRecordType) param.type).fields.containsKey(name)) {
```

Comment:
Wouldn't it be better to 
1. iterate through the param list and identify
  - all included record params of a closed record type
  - the single included record param of an open record type, if present
2. then iterate through the named args and add to the relevant records based on the expected names? 
  - if the name is present in a particular included record param, and the type is not a subtype of `never`, the field belongs to that record
  - if the name is present in a particular included record param, and the type is a subtype of `never`, the field does not belongs to that record
  - if the name is not present in a particular included record param, and it is an open record, the field belongs to that record

Reply:
Refactor the logic according to the mentioned way.